elasticsearch-transport/lib/elasticsearch/transport/transport/base.rb

```diff
@@ -96,9 +96,18 @@ def resurrect_dead_connections!
         def __rebuild_connections(arguments={})
           @hosts       = arguments[:hosts]    || []
           @options     = arguments[:options]  || {}
+          __close_connections # close old connections
```

Comment:
I think the comment is redundant -- interested people can always `git blame` and arrive on this commit, and see the link to the issue. Please make sure to include the description of the reason behind it also in the commit, so it's not only at Github...